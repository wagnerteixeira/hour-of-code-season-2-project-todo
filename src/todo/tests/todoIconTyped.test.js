import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { todoMock } from './todoMock';

configure({ adapter: new Adapter() });

import TodoIconTyped from '../todoIconTyped';

describe('Test <TodoIconTyped />', function () {

    let component;

    beforeEach(() => {
        component = shallow(
            <TodoIconTyped
                status={todoMock.defaultStatus}
            />
        );
    });

    it('render <TodoIconTyped /> ', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('render <TodoIconTyped /> ', () => {
        component = shallow(
            <TodoIconTyped
                status="TODO"
            />
        );
        expect(toJson(component).type).toEqual('ListAltIcon');
    });

    it('render <TodoIconTyped status="DOING"/> ', () => {        
        component = shallow(
            <TodoIconTyped
                status="DOING"
            />
        );
        expect(toJson(component).type).toEqual('UpdateIcon');
    });

    it('render <TodoIconTyped status="DONE"/> ', () => {        
        component = shallow(
            <TodoIconTyped
                status="DONE"
            />
        );
        expect(toJson(component).type).toEqual('CheckCircleOutlineIcon');
    });
   
});