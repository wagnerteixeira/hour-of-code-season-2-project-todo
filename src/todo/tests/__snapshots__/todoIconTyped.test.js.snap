// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TodoIconTyped /> render <TodoIconTyped />  1`] = `
<ListAltIcon
  fontSize="large"
/>
`;
